src/figuresClasses.ts

```diff
@@ -1,11 +1,89 @@
-export interface Figure {}
+enum Color {
+  red = 'red',
+  green = 'green',
+  blue = 'blue',
+}
+
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+export interface Figure {
+  shape: string;
+  color: Color;
+  getArea(): number;
+}
 
-export class Triangle implements Figure {}
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
 
-export class Circle implements Figure {}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sortSide = [this.a, this.b, this.c].sort((acc, prev) => acc - prev);
+
+    if (
+      a <= 0 ||
+      b <= 0 ||
+      c <= 0 ||
+      sortSide[2] >= sortSide[0] + sortSide[1]
+    ) {
+      throw new Error('incorrect sides');
```

Comment:
[CODE STYLE] - When throwing an error, try to provide a detailed description in the error message to make it clear what went wrong.

GOOD EXAMPLE:
throw new Error('Triangle sides must be positive numbers and satisfy the triangle inequality.');